pkg/storer/internal/pinning/pinning.go

```diff
@@ -243,6 +243,16 @@ func (c *collectionPutter) Put(ctx context.Context, st internal.Storage, writer
 		return nil
 	}
 
+	rootCollection := &pinCollectionItem{Addr: ch.Address()}
+	has, err := st.IndexStore().Has(rootCollection)
+	if err != nil {
+		return fmt.Errorf("pin store: check previous root: %w", err)
+	}
+	if has {
```

Comment:
This doesn't solve all of the problems, but only Put's that happen AFTER the initial pin is actually closed.   Consider the following:

1) Two or more simultaneous operations start pinning the exact same content.
2) They all do the individual Puts, and the root collection doesn't yet exist, so they all actually store their collectionChunks.
3) One of the pinners actually executes the Close causing the pinCollectionItem to exist.  That one "wins" and actually completes the pin.
4) The other pinners will now quit Putting their collectionChunks, but their originally stored (raced) records still exist.   And worse, their chunkstore.Put() invocation have increased the chunks' RefCnts.
5) When the other pinners finally invoke Close, they'll correctly not replace the original pinCollectionItem, but all of those incorrect RefCnt increments are still there causing those chunks to never be able to be removed from the chunkstore.

Reply:
it's been removed now